akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -273,10 +276,10 @@ private[stream] object TcpConnectionStage {
       override def onUpstreamFailure(ex: Throwable): Unit = {
         if (connection != null) {
           if (interpreter.log.isDebugEnabled) {
-            interpreter.log.debug(
-              "Aborting tcp connection because of upstream failure: {}\n{}",
-              ex.getMessage,
-              ex.getStackTrace.mkString("\n"))
+            val msg = "Aborting tcp connection to {} because of upstream failure: {}"
```

Comment:
this is the error logging improvement, it still is here, but with more info:

`[DEBUG] [12/14/2016 00:38:43.072] [ServerTest-akka.actor.default-dispatcher-4] [akka://ServerTest/user/StreamSupervisor-0/flow-1-0-unknown-operation] Aborting tcp connection to /127.0.0.1:65161 because of upstream failure:`